realm/realm-library/src/main/java/io/realm/MutableRealmObjectSchema.java

```diff
@@ -132,18 +132,18 @@ public RealmObjectSchema addRealmListField(String fieldName, RealmObjectSchema o
     }
 
     @Override
-    public RealmObjectSchema addRealmListField(String fieldName, Class<?> elementType) {
+    public RealmObjectSchema addRealmListField(String fieldName, Class<?> primitiveType) {
         checkLegalName(fieldName);
         checkFieldNameIsAvailable(fieldName);
 
-        FieldMetaData metadata = SUPPORTED_SIMPLE_FIELDS.get(elementType);
+        FieldMetaData metadata = SUPPORTED_SIMPLE_FIELDS.get(primitiveType);
         if (metadata == null) {
-            if (elementType.equals(RealmObjectSchema.class)) {
+            if (primitiveType.equals(RealmObjectSchema.class)) {
```

Comment:
I wonder if this should check against `if (primitiveType.equals(RealmObjectSchema.class) || RealmModel.class.isAssignableFrom(primitiveType)) {` to ensure that they get this error message if they provide the `Class<?>` that belongs to a RealmModel.

Currently I think they'd get `RealmList does not support lists with this type: "SomeRealmObject"` which could be confusing.

Reply:
@cmelchior please verify this

Reply:
Yes, this would be nice. Fixing